src/Module.purs

```diff
@@ -101,3 +103,18 @@ prependFolder :: Folder -> File -> File
 prependFolder (Folder folder) (File file) = File (folder <> "/" <> file)
 
 infixr 5 prependFolder as </>
+
+prependFolder' :: Folder -> File -> File
+prependFolder' (Folder folder) (File file) = File (folder <> "/" <> file <> ".fld")
+
+findM :: forall m f a b. Foldable f => f a -> (a -> AffError m (Maybe b)) -> AffError m (Maybe b)
```

Comment:
This can be given a more generic type:

`findM :: forall m f a b. Monad m => Foldable f => f a -> (a -> m (Maybe b)) -> m (Maybe b)`

However there’s a general problem with our use of `AffError`, namely that it defines a “constrained type” rather than a type, and in particular that complicates things when the `AffError` isn’t at the top level (because then the implicit quantifiers are nested).

To make this signature work, you’ll need to push through a few places where `AffError` occurs at the end of a type signature (e.g. `parseProgram`) and replace the use of `AffError` by `m` where `m` is constrained at the top level with `MonadAff m => MonadError Error m =>`. Push this through a few functions and everything should work.